rust-peer/src/protocol.rs

```diff
@@ -0,0 +1,98 @@
+use async_trait::async_trait;
```

Comment:
We should upgrade to the latest `rust-libp2p` release. This would all not be needed because we could just use the `cbor::Behaviour`!

Reply:
Yes exactly. At next PRs, I'm going to upgrade rust-libp2p and check if it works with Go and JS apps

Reply:
Note I've updated `rust-libp2p` to the latest version in the meantime :)